src/ripple/app/ledger/impl/LedgerCleaner.cpp

```diff
@@ -51,8 +51,8 @@ class LedgerCleanerImp : public LedgerCleaner
 
     std::thread thread_;
 
-    enum class State : char { readyToClean = 0, startCleaning, cleaning };
-    State state_ = State::readyToClean;
+    enum class State : char { notCleaning = 0, cleaning };
+    State state_ = State::notCleaning;
```

Comment:
`std::atomic_flag` seems to be missing the predicated `wait` overload, and because the wakeup condition is complex (stopped _or_ cleaning), I don't think we win anything by switching to `std::atomic_flag`.

Reply:
I like the simpler ```bool cleaning_{false};```